.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,62 @@
+name: PR Checks
+
+permissions:
+  contents: write
+  pull-requests: write
+
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize, reopened, ready_for_review]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  codeql:
+    name: Run CodeQL Scan
+    runs-on: ubuntu-latest
+    concurrency: ${{ github.workflow }}-${{ matrix.language }}-${{ github.ref }}
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    strategy:
+      fail-fast: false
+      matrix:
+        language:
+            - javascript
+    steps:
+      - name: Run Code Scanning
+        uses: department-of-veterans-affairs/codeql-tools/codeql-analysis@main
+        with:
+          language: ${{ matrix.language }}
+
+  security:
+    name: NPM Audit
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Run NPM audit
+        run: npm audit --production --audit-level high
+
+  auto-merge:
+    name: Auto-Merge Dependabot PRs
+    needs: [codeql, security]
```

Comment:
I took it as given that we wouldn't want to merge a Dependabot PR until other PR checks had passed.

Making sure that this new job would wait until other checks finished was tricky when other checks were defined in different files, because it's not possible to reference jobs outside of the current workflow in `needs`.

I found some awkward ways to work around that (by checking for an artifact file published by another workflow or switching to a workflow_dispatch trigger instead of a pull_request trigger, for example), but they all seemed at least a little convoluted.

Instead, I've taken the approach of just combining the existing PR checks into a single file, so that they can be referenced easily amongst each other. If this is undesirable I can try those other workarounds again, though, just let me know.